src/Resolvers/TransformedDataCollectableResolver.php

```diff
@@ -75,11 +75,10 @@ protected function transformItems(
         bool $executeWrap,
         TransformationContext $nestedContext,
     ): array {
-        $collection = [];
-
-        foreach ($items as $key => $value) {
-            $collection[$key] = $this->transformationClosure($nestedContext)($value);
-        }
+        $collection = array_map(
+            fn ($value) => $this->transformationClosure($nestedContext)($value),
+            is_array($items) ? $items : $items->all()
+        );
```

Comment:
Same as above, we want the performance of foreach

Reply:
> Same as above, we want the performance of foreach

See https://github.com/spatie/laravel-data/pull/920#discussion_r1921085756.